packages/gnosis-guild-app/src/hooks/useContribute.ts

```diff
@@ -136,19 +136,27 @@ export const useContribute = (): Contribution => {
     }
     setModalFooter("Cancelling Subscription...");
     setContributeLoading(true);
-    const tx = await unsubscribe(
-      subscriber.keyId.toString(),
-      guildMetadata.id,
-      ethersProvider
-    );
+    try {
+      const tx = await unsubscribe(
+        subscriber.keyId.toString(),
+        guildMetadata.id,
+        ethersProvider
+      );
 
-    if (tx) {
-      await tx.wait();
+      if (tx) {
+        await tx.wait();
+      }
+    } catch (err) {
+      enqueueSnackbar("Failed to cancel guild contribution", {
+        anchorOrigin: { horizontal: "right", vertical: "top" },
+        preventDuplicate: true,
+        variant: "error",
+      });
+      console.error(err);
     }
 
     setContributeLoading(false);
     setSubscribed(false);
-    setSubscriber();
```

Comment:
@santteegt I saw you added this? Do we need it? If it is there then when a user cancels the button does not show submit

Reply:
your change looks good